nifi-nar-bundles/nifi-salesforce-bundle/nifi-salesforce-processors/src/test/java/org/apache/nifi/processors/salesforce/util/SalesforceConfigAware.java

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.salesforce.util;
+
+import org.apache.nifi.oauth2.StandardOauth2AccessTokenProvider;
+import org.apache.nifi.reporting.InitializationException;
+import org.apache.nifi.util.TestRunner;
+
+/**
+ * Set the following constants:<p>
+ * VERSION<p>
+ * BASE_URL<p>
+ * USERNAME<p>
+ * PASSWORD<p>
+ * CLIENT_ID<p>
+ * CLIENT_SECRET<p>
+ */
+public interface SalesforceConfigAware {
+    String VERSION = "46.0";
+    String BASE_URL = "https://cloudera-2a-dev-ed.my.salesforce.com";
+
+    String AUTHORIZATION_SERVER_URL = BASE_URL + "/services/oauth2/token";
+    String USERNAME = "???";
+    String PASSWORD = "???";
+    String CLIENT_ID = "???";
+    String CLIENT_SECRET = "???";
```

Comment:
What do you think about removing these hard-coded placeholders and instead retrieving the values from environment variables or system properties?

Reply:
Hi @exceptionfactory,
This is a collaboration work between me and @Lehel44 and I made this part.

How we handle manual tests currently I would say this is the more convenient approach.

Anyone who wants to run these tests needs to add these settings manually one way or another.
As long as we don't have a more standardized way of providing configuration details to tests (so that they can become automatic) simply editing the classes themselves is the simplest and thus gives the best user experience.

Reply:
Thanks for the reply @tpalfy, I'm fine with leaving this as-is for integration testing.  In the future it may be useful to consider a larger refactor of integration tests to support configuration through external variables - which would enable automated execution with custom configuration values.  For works for now.